AzureData/Source/Resources/ResourceOracle.swift

```diff
@@ -98,15 +98,43 @@ public class ResourceOracle {
             selfLinkLookup[altLink] = nil
         }
     }
-    
+
+    fileprivate static func _removeLinks(forResourceWithLink link: String) {
+
+        if let selfLink = selfLinkLookup[link] {
+            selfLinkLookup[link] = nil
+
+            if let _ = altLinkLookup[selfLink] {
+                altLinkLookup[selfLink] = nil
+            }
+
+            return
+        }
+
+        if let altLink = altLinkLookup[link] {
+            altLinkLookup[link] = nil
+
+            if let _ = selfLinkLookup[altLink] {
```

Comment:
I don't think it's necessary to check for the existence of the key before attempting to remove it via subscript